test/parser/LiteralOrIriTest.cpp

```diff
@@ -21,6 +22,57 @@ TEST(IriTest, IriCreation) {
               asStringViewUnsafe(iri.getContent()));
 }
 
+TEST(IriTest, getBaseIri) {
+  Iri iri1 = Iri::fromIriref("<http://purl.uniprot.org/uniprot/>");
+  EXPECT_THAT("<http://purl.uniprot.org/>",
```

Comment:
The idiomatic gtest way (including the good error messages) to get rid of the duplication, is to use a matcher:
```
auto baseMatcher = [](const std::string& expected) {
  // Check that toStringRepresentation() returns a `const string&`, otherwise this might dangle.
  auto getStringOfBase = [](const std::string& iriref) { return Iri::fromIriref(iriref).getBaseiri().toStringRepresentation();};
  return ::testing::ResultOf(getStringOfBase, expected);
}
EXPECT_THAT("<http://purl.uniprot.org/somethingRelativeElse>", baseMatcher("<http://purl.uniprot.org/>");
```
(+- details).

Reply:
Or you do it similarly to how you did it below.